main/acle.md

```diff
@@ -9382,6 +9415,2344 @@ are named after. All of the functions have external linkage.
+  
+  
+  // Variants are available for:
+  //   _za32[_s32]
+  //   _za32[_u32]
+  //   _za64[_s64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svadd_write[_single]_za32[_s32]_vg1x4(uint32_t slice_base,
+                                             uint64_t slice_offset,
+                                             svint32x4_t zn, svint32_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_u32]
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_write[_single]_za32[_u32]_vg1x2(uint32_t slice_base,
+                                             uint64_t slice_offset,
+                                             svuint32x2_t zn, svuint32_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_u32]
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_write[_single]_za32[_u32]_vg1x4(uint32_t slice_base,
+                                             uint64_t slice_offset,
+                                             svuint32x4_t zn, svuint32_t zm);
+  
+  ```
+
+#### ADD, SUB (store into ZA, multi)
+
+Multi-vector add/sub, storing into ZA
+
+The additional '_write' suffix indicates that the operation is not accumulating,
+ the result is written directly into ZA.
+
+``` c
+  // Variants are available for:
+  //   _za32[_s32]
+  //   _za32[_u32]
+  //   _za64[_s64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svadd_write_za32[_s32]_vg1x2(uint32_t slice_base, uint64_t slice_offset,
+                                    svint32x2_t zn, svint32x2_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_s32]
+  //   _za32[_u32]
+  //   _za64[_s64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svadd_write_za32[_s32]_vg1x4(uint32_t slice_base, uint64_t slice_offset,
+                                    svint32x4_t zn, svint32x4_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_u32]
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_write_za32[_u32]_vg1x2(uint32_t slice_base, uint64_t slice_offset,
+                                    svuint32x2_t zn, svuint32x2_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_u32]
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_write_za32[_u32]_vg1x4(uint32_t slice_base, uint64_t slice_offset,
+                                    svuint32x4_t zn, svuint32x4_t zm);
+  
+  ```
+
+#### ADD, SUB (vectors)
+
+Multi-vector add/sub
```

Comment:
There is only assembly for ADD vectors.
Is that correct that the ACLE should also implement sub.
Even if there is no assembly instructions to map the intrinsic to?

Reply:
For the SVE ACLE we've shied away from doing this, because it hides a performance impact i.e. that the multi-vector ADD is one instruction, whereas SUB is multiple. I also don't believe there are any multi-vector negate instructions, which would make the performance difference more remarkable.

I should remove the `, SUB` from the section title here though, thanks for pointing that out!

Reply:
Resolved in https://github.com/ARM-software/acle/pull/217/commits/18c5da294a88c3a4fbb4c8d3301c5f71293f1aff